### ISSUE TITLE
Error when Indigo remote sensor doesn't exist in Ecobee data

### ISSUE THREAD

**Issue by CommanderB** [FlyingDiver/Indigo-Ecobee-2#23]
2.4.1 and 2.4.0 give following error

`Ecobee 2 Error                  Error in plugin execution runConcurrentThread:

Traceback (most recent call last):
  File "plugin.py", line 210, in runConcurrentThread
  File "/Library/Application Support/Perceptive Automation/Indigo 2021.1/Plugins/Ecobee 2.indigoPlugin/Contents/Server Plugin/ecobee_devices.py", line 369, in update
  File "/Library/Application Support/Perceptive Automation/Indigo 2021.1/Plugins/Ecobee 2.indigoPlugin/Contents/Server Plugin/temperature_scale.py", line 32, in convertFromEcobee
ValueError: could not convert string to float: unknown`

Thermostat data (note two homes in Ecobee)

```
Ecobee 2                        {
    "311000124293": {
        "actualHumidity": 65,
        "actualTemperature": 757,
        "brand": "ecobee",
        "climates": {
            "away": "Away",
            "home": "Home",
            "sleep": "Sleep"
        },
        "currentClimate": "home",
        "desiredCool": 755,
        "desiredFanMode": "auto",
        "desiredHeat": 705,
        "equipmentStatus": "",
        "fanMinOnTime": 0,
        "features": "Home",
        "hvacMode": "cool",
        "internal": {
            "humidity": "65",
            "occupancy": "false",
            "temperature": "755"
        },
        "latestEventType": "template",
        "modelNumber": "athenaSmart",
        "name": "Dining Room",
        "remotes": {
            "DD63": {
                "name": "Kitchen",
                "occupancy": "true",
                "temperature": "760",
                "thermostat": "311000124293"
            },
            "F2ZC": {
                "name": "Office",
                "occupancy": "false",
                "temperature": "754",
                "thermostat": "311000124293"
            }
        }
    },
    "312975924798": {
        "actualHumidity": 71,
        "actualTemperature": 772,
        "brand": "ecobee",
        "climates": {
            "away": "Away",
            "home": "Home",
            "sleep": "Sleep",
            "smart1": "Pre Bed Cool"
        },
        "currentClimate": "away",
        "desiredCool": 824,
        "desiredFanMode": "auto",
        "desiredHeat": 644,
        "equipmentStatus": "",
        "fanMinOnTime": 0,
        "features": "Home",
        "hvacMode": "cool",
        "internal": {
            "humidity": "71",
            "occupancy": "true",
            "temperature": "740"
        },
        "latestEventType": "template",
        "modelNumber": "athenaSmart",
        "name": "Landing",
        "remotes": {
            "DD6D": {
                "name": "F Bedroom",
                "occupancy": "true",
                "temperature": "764",
                "thermostat": "312975924798"
            },
            "DYTG": {
                "name": "Master Bedroom",
                "occupancy": "true",
                "temperature": "780",
                "thermostat": "312975924798"
            },
            "F2KW": {
                "name": "P Bedroom",
                "occupancy": "false",
                "temperature": "775",
                "thermostat": "312975924798"
            }
        }
    },
    "316593012862": {
        "actualHumidity": 62,
        "actualTemperature": 759,
        "brand": "ecobee",
        "climates": {
            "away": "Away",
            "home": "Home",
            "sleep": "Sleep"
        },
        "currentClimate": "home",
        "desiredCool": 761,
        "desiredFanMode": "auto",
        "desiredHeat": 716,
        "equipmentStatus": "",
        "fanMinOnTime": 0,
        "features": "Home,HomeKit",
        "hvacMode": "cool",
        "internal": {
            "humidity": "62",
            "occupancy": "false",
            "temperature": "758"
        },
        "latestEventType": "template",
        "modelNumber": "athenaSmart",
        "name": "Top Bedroom",
        "remotes": {
            "3MVH": {
                "name": "Nics Desk",
                "occupancy": "false",
                "temperature": "802",
                "thermostat": "316593012862"
            }
        }
    },
    "317437217236": {
        "actualHumidity": 84,
        "actualTemperature": 616,
        "brand": "ecobee",
        "climates": {
            "away": "Away",
            "home": "Home",
            "sleep": "Sleep"
        },
        "currentClimate": "away",
        "desiredCool": 4000,
        "desiredFanMode": "auto",
        "desiredHeat": 554,
        "equipmentStatus": "",
        "fanMinOnTime": 0,
        "features": "Home,HomeKit",
        "hvacMode": "off",
        "internal": {
            "humidity": "84",
            "occupancy": "false",
            "temperature": "616"
        },
        "latestEventType": "vacation",
        "modelNumber": "athenaSmart",
        "name": "Basement",
        "remotes": {}
    },
    "411912730332": {
        "actualHumidity": 50,
        "actualTemperature": 722,
        "brand": "ecobee",
        "climates": {
            "away": "Away",
            "home": "Home",
            "sleep": "Sleep"
        },
        "currentClimate": "away",
        "desiredCool": 4000,
        "desiredFanMode": "auto",
        "desiredHeat": 554,
        "equipmentStatus": "",
        "fanMinOnTime": 0,
        "features": "Home,HomeKit",
        "hvacMode": "heat",
        "internal": {
            "humidity": "50",
            "temperature": "731"
        },
        "latestEventType": "vacation",
        "modelNumber": "nikeSmart",
        "name": "Upstairs",
        "remotes": {
            "CYKP": {
                "name": "Master Bedroom",
                "occupancy": "false",
                "temperature": "718",
                "thermostat": "411912730332"
            },
            "JWF8": {
                "name": "Bunk Room",
                "occupancy": "false",
                "temperature": "727",
                "thermostat": "411912730332"
            }
        }
    },
    "411932444266": {
        "actualHumidity": 53,
        "actualTemperature": 716,
        "brand": "ecobee",
        "climates": {
            "away": "Away",
            "home": "Home",
            "sleep": "Sleep"
        },
        "currentClimate": "away",
        "desiredCool": 4000,
        "desiredFanMode": "auto",
        "desiredHeat": 554,
        "equipmentStatus": "",
        "fanMinOnTime": 0,
        "features": "Home,HomeKit",
        "hvacMode": "heat",
        "internal": {
            "humidity": "53",
            "temperature": "721"
        },
        "latestEventType": "vacation",
        "modelNumber": "nikeSmart",
        "name": "Bathrooms",
        "remotes": {
            "BL75": {
                "name": "Master Bathroom",
                "occupancy": "false",
                "temperature": "710",
                "thermostat": "411932444266"
            },
            "BN2G": {
                "name": "Small Bathroom",
                "occupancy": "false",
                "temperature": "720",
                "thermostat": "411932444266"
            }
        }
    },
    "411998799739": {
        "actualHumidity": 69,
        "actualTemperature": 723,
        "brand": "ecobee",
        "climates": {
            "away": "Away",
            "home": "Home",
            "sleep": "Sleep"
        },
        "currentClimate": "away",
        "desiredCool": 824,
        "desiredFanMode": "auto",
        "desiredHeat": 644,
        "equipmentStatus": "",
        "fanMinOnTime": 0,
        "features": "Home,HomeKit",
        "hvacMode": "off",
        "internal": {
            "humidity": "69",
            "temperature": "723"
        },
        "latestEventType": null,
        "modelNumber": "nikeSmart",
        "name": "Den",
        "remotes": {
            "JWDZ": {
                "name": "Den Sensor",
                "occupancy": "false",
                "temperature": "unknown",
                "thermostat": "411998799739"
            }
        }
    },
    "511824987795": {
        "actualHumidity": 68,
        "actualTemperature": 709,
        "brand": "ecobee",
        "climates": {
            "away": "Away",
            "home": "Home",
            "sleep": "Sleep"
        },
        "currentClimate": "away",
        "desiredCool": 824,
        "desiredFanMode": "auto",
        "desiredHeat": 644,
        "equipmentStatus": "",
        "fanMinOnTime": 0,
        "features": "Home,HomeKit",
        "hvacMode": "off",
        "internal": {
            "humidity": "68",
            "occupancy": "false",
            "temperature": "709"
        },
        "latestEventType": null,
        "modelNumber": "apolloSmart",
        "name": "Nics Study",
        "remotes": {}
    },
    "511856028154": {
        "actualHumidity": 64,
        "actualTemperature": 746,
        "brand": "ecobee",
        "climates": {
            "away": "Away",
            "home": "Home",
            "sleep": "Sleep"
        },
        "currentClimate": "home",
        "desiredCool": 749,
        "desiredFanMode": "auto",
        "desiredHeat": 699,
        "equipmentStatus": "",
        "fanMinOnTime": 0,
        "features": "Home,HomeKit",
        "hvacMode": "cool",
        "internal": {
            "humidity": "64",
            "occupancy": "true",
            "temperature": "745"
        },
        "latestEventType": "template",
        "modelNumber": "apolloSmart",
        "name": "Family Room",
        "remotes": {
            "DZKD": {
                "name": "FR Wall",
                "occupancy": "false",
                "temperature": "748",
                "thermostat": "511856028154"
            }
        }
    },
    "511877273997": {
        "actualHumidity": 60,
        "actualTemperature": 687,
        "brand": "ecobee",
        "climates": {
            "away": "Away",
            "home": "Home",
            "sleep": "Sleep"
        },
        "currentClimate": "away",
        "desiredCool": 4000,
        "desiredFanMode": "auto",
        "desiredHeat": 554,
        "equipmentStatus": "",
        "fanMinOnTime": 0,
        "features": "Home,HomeKit",
        "hvacMode": "heat",
        "internal": {
            "humidity": "60",
            "occupancy": "false",
            "temperature": "687"
        },
        "latestEventType": "vacation",
        "modelNumber": "apolloSmart",
        "name": "Back Study",
        "remotes": {}
    },
    "511880234590": {
        "actualHumidity": 63,
        "actualTemperature": 680,
        "brand": "ecobee",
        "climates": {
            "away": "Away",
            "home": "Home",
            "sleep": "Sleep"
        },
        "currentClimate": "away",
        "desiredCool": 4000,
        "desiredFanMode": "auto",
        "desiredHeat": 554,
        "equipmentStatus": "",
        "fanMinOnTime": 0,
        "features": "Home,HomeKit",
        "hvacMode": "heat",
        "internal": {
            "humidity": "63",
            "occupancy": "false",
            "temperature": "670"
        },
        "latestEventType": "vacation",
        "modelNumber": "apolloSmart",
        "name": "Living Room",
        "remotes": {
            "DYR2": {
                "name": "Kitchen",
                "occupancy": "false",
                "temperature": "690",
                "thermostat": "511880234590"
            }
        }
    },
    "511886904338": {
        "actualHumidity": 60,
        "actualTemperature": 671,
        "brand": "ecobee",
        "climates": {
            "away": "Away",
            "home": "Home",
            "sleep": "Sleep"
        },
        "currentClimate": "away",
        "desiredCool": 4000,
        "desiredFanMode": "auto",
        "desiredHeat": 554,
        "equipmentStatus": "",
        "fanMinOnTime": 0,
        "features": "Home,HomeKit",
        "hvacMode": "heat",
        "internal": {
            "humidity": "60",
            "occupancy": "false",
            "temperature": "670"
        },
        "latestEventType": "vacation",
        "modelNumber": "apolloSmart",
        "name": "Guest Room",
        "remotes": {
            "GFD5": {
                "name": "Persis Room",
                "occupancy": "false",
                "temperature": "673",
                "thermostat": "511886904338"
            }
        }
    }
}
   Ecobee 2                        {
    "3MVH": {
        "name": "Nics Desk",
        "occupancy": "false",
        "temperature": "802",
        "thermostat": "316593012862"
    },
    "BL75": {
        "name": "Master Bathroom",
        "occupancy": "false",
        "temperature": "710",
        "thermostat": "411932444266"
    },
    "BN2G": {
        "name": "Small Bathroom",
        "occupancy": "false",
        "temperature": "720",
        "thermostat": "411932444266"
    },
    "CYKP": {
        "name": "Master Bedroom",
        "occupancy": "false",
        "temperature": "718",
        "thermostat": "411912730332"
    },
    "DD63": {
        "name": "Kitchen",
        "occupancy": "true",
        "temperature": "760",
        "thermostat": "311000124293"
    },
    "DD6D": {
        "name": "F Bedroom",
        "occupancy": "true",
        "temperature": "764",
        "thermostat": "312975924798"
    },
    "DYR2": {
        "name": "Kitchen",
        "occupancy": "false",
        "temperature": "690",
        "thermostat": "511880234590"
    },
    "DYTG": {
        "name": "Master Bedroom",
        "occupancy": "true",
        "temperature": "780",
        "thermostat": "312975924798"
    },
    "DZKD": {
        "name": "FR Wall",
        "occupancy": "false",
        "temperature": "748",
        "thermostat": "511856028154"
    },
    "F2KW": {
        "name": "P Bedroom",
        "occupancy": "false",
        "temperature": "775",
        "thermostat": "312975924798"
    },
    "F2ZC": {
        "name": "Office",
        "occupancy": "false",
        "temperature": "754",
        "thermostat": "311000124293"
    },
    "GFD5": {
        "name": "Persis Room",
        "occupancy": "false",
        "temperature": "673",
        "thermostat": "511886904338"
    },
    "JWDZ": {
        "name": "Den Sensor",
        "occupancy": "false",
        "temperature": "unknown",
        "thermostat": "411998799739"
    },
    "JWF8": {
        "name": "Bunk Room",
        "occupancy": "false",
        "temperature": "727",
        "thermostat": "411912730332"
    }
}
```

**Comment by FlyingDiver** [comment]
Do you get the same error when set to F?

Can you turn on debug logging and capture the log just before the error?

**Comment by CommanderB** [comment]
So it happens the same with Celsius and Fahrenheit per below. Doesnt occur with 2.3.5, does with 2.4.0 and 2.4.1. I will poke about the Den thermostat to see if there is something set differently to others. 

Celsius - 

   ```
Ecobee 2 Debug                  Ecobee Sensor Top Study: Updating device
   Ecobee 2 Debug                  Ecobee Sensor Top Study: Ecobee Account device assigned, 1686082423
   Ecobee 2 Debug                  update: no remote sensor data found for address BN2P
   Ecobee 2 Debug                  Ecobee Sensor Bunk Room (JWF8): Updating device
   Ecobee 2 Debug                  Ecobee Sensor Bunk Room (JWF8): Ecobee Account device assigned, 1686082423
   Ecobee 2 Debug                  Ecobee Sensor Bunk Room (JWF8): Reported temp: 732, converted temp: 22.8888888889
   Ecobee 2 Debug                  Ecobee Sensor Den: Updating device
   Ecobee 2 Debug                  Ecobee Sensor Den: Ecobee Account device assigned, 1686082423
   Ecobee 2 Error                  Error in plugin execution runConcurrentThread:

Traceback (most recent call last):
  File "plugin.py", line 210, in runConcurrentThread
  File "/Library/Application Support/Perceptive Automation/Indigo 2021.1/Plugins/Ecobee 2.indigoPlugin/Contents/Server Plugin/ecobee_devices.py", line 369, in update
  File "/Library/Application Support/Perceptive Automation/Indigo 2021.1/Plugins/Ecobee 2.indigoPlugin/Contents/Server Plugin/temperature_scale.py", line 32, in convertFromEcobee
ValueError: could not convert string to float: unknown

   Ecobee 2 Error                  plugin runConcurrentThread function returned or failed (will attempt again in 10 seconds)
   Ecobee 2 Debug                  runConcurrentThread starting

```

Fahrenheit - 

   ```
Ecobee 2 Debug                  Ecobee Top Bedroom Nics Desk: Updating device
   Ecobee 2 Debug                  Ecobee Top Bedroom Nics Desk: Reported temp: 776, converted temp: 77.6
   Ecobee 2 Debug                  Ecobee Sensor Top Study: Updating device
   Ecobee 2 Debug                  update: no remote sensor data found for address BN2P
   Ecobee 2 Debug                  Ecobee Sensor Bunk Room (JWF8): Updating device
   Ecobee 2 Debug                  Ecobee Sensor Bunk Room (JWF8): Reported temp: 732, converted temp: 73.2
   Ecobee 2 Debug                  Ecobee Sensor Den: Updating device
   Ecobee 2 Error                  Error in plugin execution runConcurrentThread:

Traceback (most recent call last):
  File "plugin.py", line 210, in runConcurrentThread
  File "/Library/Application Support/Perceptive Automation/Indigo 2021.1/Plugins/Ecobee 2.indigoPlugin/Contents/Server Plugin/ecobee_devices.py", line 369, in update
  File "/Library/Application Support/Perceptive Automation/Indigo 2021.1/Plugins/Ecobee 2.indigoPlugin/Contents/Server Plugin/temperature_scale.py", line 20, in convertFromEcobee
ValueError: could not convert string to float: unknown

   Ecobee 2 Error                  plugin runConcurrentThread function returned or failed (will attempt again in 10 seconds)
   Ecobee 2 Debug                  runConcurrentThread starting
```

**Comment by FlyingDiver** [comment]
Try deleting the remote sensor BN2P.

**Comment by CommanderB** [comment]
Yup, that's just what I found. In the Ecobee app that sensor is showing as unavailable

**Comment by CommanderB** [comment]
So the unavailable Den sensor was JWDZ, I assumed thats what was erroring, but it wasnt as you knew and deleting that had no effect.

There was a Top Bedroom sensor BN2P which had been removed from the thermostat some time ago, but I had not removed from Indigo. Removing from Indigo cleared the error. 

Thank you!

**Comment by FlyingDiver** [comment]
I’ll need to update the code so it doesn’t error when that happens.

**Comment by FlyingDiver** [comment]
Fixed in 2.4.2
